glide-core/tests/test_client.rs

```diff
@@ -540,6 +561,83 @@ pub(crate) mod shared_client_tests {
         });
     }
 
+    #[test]
+    #[serial_test::serial]
+    fn test_client_telemetry_standalone() {
+        Telemetry::reset();
+        block_on_all(async move {
+            // create a server with 2 clients
+            let server_config = TestConfiguration {
+                use_tls: false,
+                ..Default::default()
+            };
+
+            let test_basics = utilities::setup_test_basics_internal(&server_config).await;
+            let server = BackingServer::Standalone(test_basics.server);
+
+            // setup_test_basics_internal internally, starts a single client connection
+            assert_eq!(Telemetry::total_connections(), 1);
+            assert_eq!(Telemetry::total_clients(), 1);
+
+            {
+                // Create 2 more clients, confirm that they are tracked
+                let _client1 = create_client(&server, server_config.clone()).await;
+                let _client2 = create_client(&server, server_config).await;
+
+                // Each client maintains a single connection
+                assert_eq!(Telemetry::total_connections(), 3);
+                assert_eq!(Telemetry::total_clients(), 3);
+
+                // Connections are dropped here
+            }
+
+            // Confirm 1 connection & client remain
+            assert_eq!(Telemetry::total_connections(), 1);
+            assert_eq!(Telemetry::total_clients(), 1);
+        });
+    }
+
+    #[test]
+    #[serial_test::serial]
+    fn test_client_telemetry_cluster() {
```

Comment:
what about different scenarios - when connections are dropped, when all cluster is dropped and reconnect_from_initial_nodes is called, when new node is being added

Reply:
Will be done in a different PR (this PR is the first in a set of PRs). In the next PR, I will focus on adding more capabilities to the `cluster_manager.py` script (e.g. the ability to terminate shard / primary in a cluster etc)